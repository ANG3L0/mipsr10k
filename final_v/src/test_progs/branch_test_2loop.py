if __name__ == '__main__':
	x = 1015
	y = 4
	z = 2
	for i in range(10):
		for j in range(5):
			x = x + i*j
			y = y + i + j
			z = z + z
	print x
	print y
	print z
