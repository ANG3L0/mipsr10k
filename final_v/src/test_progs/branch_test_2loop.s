

        lda  $r1, 0xa 
        lda  $r2, 0x1 
        lda  $r3, 0x3f7
        lda  $r4, 0x4
        lda  $r10, 0x2
loop2:  subq $r1, $r2, $r1  
        lda  $r5, 5
loop1:  subq $r5, $r2, $r5
	mulq $r1, $r5, $r6
	addq $r6, $r3, $r3
	addq $r1, $r5, $r7
	addq $r7, $r4, $r4
        addq $r10, $r10, $r10 
        bne  $r5, loop1
        bne  $r1, loop2
	addq $r1, $r2, $r8
	addq $r5, $r2, $r9
        call_pal 0x555
