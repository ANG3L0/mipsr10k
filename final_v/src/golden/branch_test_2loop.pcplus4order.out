4:lda
8:lda
12:lda
16:lda
20:lda
24:subq
28:lda
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
60:bne
24:subq
28:lda
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
60:bne
24:subq
28:lda
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
60:bne
24:subq
28:lda
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
60:bne
24:subq
28:lda
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
60:bne
24:subq
28:lda
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
60:bne
24:subq
28:lda
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
60:bne
24:subq
28:lda
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
60:bne
24:subq
28:lda
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
60:bne
24:subq
28:lda
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
32:subq
36:mulq
40:addq
44:addq
48:addq
52:addq
56:bne
60:bne
64:addq
68:addq
72:halt
